src/utils.rs

```diff
@@ -37,7 +37,10 @@ impl Graphics {
         else {
             dst = Point2::new(coords.x, coords.y);
         }
-        draw(_ctx, &mut graphics_text, dst, 0.0)?;
+        let previous_color = graphics::get_color(_ctx);      // store previous color
```

Comment:
I think it would be better to explain why we store the previous color instead of commenting each statement with what it is doing. Just my two cents.

Reply:
Fixed!